flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/RangePathParameter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages;
+
+import org.apache.flink.runtime.rest.handler.legacy.files.FileOffsetRange;
+
+/**
+ * Path parameter identifying the offset range of a log file to read.
+ */
+public class RangePathParameter extends MessagePathParameter<FileOffsetRange> {
```

Comment:
Good idea!

Reply:
`/taskmanagers/:%s/log?filename=xxx&start=123&size=456` may be better since it is compatible with the old API `/taskmanagers/:%s/log`, in which `filename`, `start` and `size` are all optional.